### PR TITLE
Added ContentProvider to resolve calendar event models to URLs

### PR DESCRIPTION
**Pull request by aschempp**
This will allow the router to generate URLs for calendar event models.

**Dependencies:**
- [ ] https://github.com/contao/contao/pull/2031

### CODE REVIEW

File: calendar-bundle/src/Resources/contao/classes/Events.php [hunk 1]
```diff
@@ -423,73 +419,25 @@ protected function addEvent($objEvents, $intStart, $intEnd, $intBegin, $intLimit
 	 * @param boolean             $blnAbsolute
 	 *
 	 * @return string
+	 *
+	 * @deprecated Deprecated since Contao 4.10. Use the Symfony router instead.
 	 */
 	public static function generateEventUrl($objEvent, $blnAbsolute=false)
 	{
-		$strCacheKey = 'id_' . $objEvent->id . ($blnAbsolute ? '_absolute' : '');
+		@trigger_error(__METHOD__.' is deprecated since Contao 4.10. Use the Symfony router instead.', E_USER_DEPRECATED);
```
Comment (bytehead) [inline calendar-bundle/src/Resources/contao/classes/Events.php]: Use `trigger_deprecation` here (see #1946).

**Review by bytehead — CHANGES_REQUESTED** [review]

File: calendar-bundle/src/Resources/contao/classes/Events.php (hunk 1 above)
Comment (aschempp) [reply calendar-bundle/src/Resources/contao/classes/Events.php:427]: Fixed in 5db43fe2d7f72fb92f3e92804f74419d6fe17ba8

Comment (aschempp) [reply calendar-bundle/src/Resources/contao/classes/Events.php]: fixed in 7cbf8f03f78304f8e46128b5ab57b60bcd36a45e

**Comment by bytehead** [comment]
What about tests for the routing provider?

**Comment by leofeyer** [comment]
It seems you forgot to adjust the `PreviewUrlConverterListenerTest` class. And you should add a unit test for the `EventsRouteProvider` class. 🙈

**Pull request closed without merging**